students/Dkuchan/ClassProject/PythonClassProject.py

```diff
@@ -0,0 +1,625 @@
+# PythonClassProject.py
+''' This python program is written for the advanced python programming class
+    it is a scouting program written for the First Robotics Competition based
+    on the 2017 game FIRST Steamworks. '''
+
+
+
+
+'''Need to add a tuple for allowed users to baseline who does what'''
+
+
+
+import math
+import os
+import csv
+import random
+
+matchlist = []
+teamlist = []   #HOLDS THE TEAM NUMBERS BECAUSE I COULD NOT GET A DIRECT LIST CREATION TO WORK
+teamlist2 = []  #HOLDS THE TEAM OBJECTS
+montelist = []
+teamdict = {}
+monteblue = [0, 0, 0]
+montered = [0, 0, 0]
+
+
+class Match:
+
+    # A match is a collection of data which represents a match FOR A SINGLE TEAM.
+    # Match is the primary class which is edited by the user.
+    # it is important to note that this is intended to log a section of a match and 
+    # a single instance does not contain the entire dataset for a real life match.
+    # The reason for this is because a single person will not 
+    # log more than one team in a match.  
+
+    # bool winner - Done
+    # match number - Done
+    # team being tracked - Done
+    # alliance color - Done
+    # team position - Done
+    # bool pilot in airship - Done
+    # gears delivered - Done
+    # KPA scored - Done
+    # bool gear delivered in auto - Done
+    # bool climb - Done
+    # KPA in auto - Done
+
+    def __init__(self, mnumber, teamnumber):
+        self.matchnumber = mnumber
+        self.team = teamnumber
+        self.win = 1
+        self.gears = None
+        self.alliance = None
+        self.fieldpos = None
+        self.pilot = None
+        self.autogears = None
+        self.kpa = None
+        self.autokpa = None
+        self.climb = None
+        self.penalties = None
+
+    def __repr__(self):
+        return '{}, {}: {}'.format(self.team, self.matchnumber, self.win)
+
+    def __str__(self):
+        return 'Match {}, Team {}'.format(self.matchnumber, self.team)
+
+    #def __iter__(self):
+
+class MonteResult(Match):
+    # this class is intended to contain results for monte carlo runs
+    # it is a sublcass of Match as a monte run is just simulated Matches
+    # functions of match are not all used at this point.  perhaps with future updates
+
+    def __init__(self, bluescore, redscore, runnum):
```

Comment:
you haven't initialised the Match class -- so this won't have all the attributes of a Match. If this works, then this really isn't a subclass...